libs/vm-modules/src/math/bignumber.cpp

```diff
@@ -250,6 +258,116 @@ bool UInt256Wrapper::FromJSON(JSONVariant const &variant)
   return true;
 }
 
+void UInt256Wrapper::Add(fetch::vm::Ptr<Object> &lhso, fetch::vm::Ptr<Object> &rhso)
+{
+  Ptr<UInt256Wrapper> lhs = lhso;
+  Ptr<UInt256Wrapper> rhs = rhso;
+  if (lhs->IsTemporary())
+  {
+    lhs->number_ += rhs->number_;
+    return;
+  }
+  if (rhs->IsTemporary())
+  {
+    rhs->number_ += lhs->number_;
+    lhso = std::move(rhs);
+    return;
+  }
+  Ptr<UInt256Wrapper> n(
+      new UInt256Wrapper(vm_, fetch::vm::TypeIds::UInt256, lhs->number_ + rhs->number_));
+  lhso = std::move(n);
+}
+
+void UInt256Wrapper::Subtract(fetch::vm::Ptr<Object> &lhso, fetch::vm::Ptr<Object> &rhso)
+{
+  Ptr<UInt256Wrapper> lhs = lhso;
+  Ptr<UInt256Wrapper> rhs = rhso;
+  if (lhs->IsTemporary())
+  {
+    lhs->number_ -= rhs->number_;
+    return;
+  }
+  if (rhs->IsTemporary())
+  {
+    rhs->number_ -= lhs->number_;
+    lhso = std::move(rhs);
+    return;
+  }
+  Ptr<UInt256Wrapper> n(
+      new UInt256Wrapper(vm_, fetch::vm::TypeIds::UInt256, lhs->number_ - rhs->number_));
+  lhso = std::move(n);
+}
+
+void UInt256Wrapper::InplaceAdd(fetch::vm::Ptr<Object> const &lhso,
+                                fetch::vm::Ptr<Object> const &rhso)
+{
+  Ptr<UInt256Wrapper> lhs = lhso;
+  Ptr<UInt256Wrapper> rhs = rhso;
+  lhs->number_ += rhs->number_;
+}
+
+void UInt256Wrapper::InplaceSubtract(fetch::vm::Ptr<Object> const &lhso,
+                                     fetch::vm::Ptr<Object> const &rhso)
+{
+  Ptr<UInt256Wrapper> lhs = lhso;
+  Ptr<UInt256Wrapper> rhs = rhso;
+  lhs->number_ -= rhs->number_;
+}
+
+void UInt256Wrapper::Multiply(Ptr<Object> &lhso, Ptr<Object> &rhso)
+{
+  Ptr<UInt256Wrapper> lhs = lhso;
+  Ptr<UInt256Wrapper> rhs = rhso;
+  if (lhs->IsTemporary())
+  {
+    lhs->number_ *= rhs->number_;
+    return;
+  }
+  if (rhs->IsTemporary())
+  {
+    rhs->number_ *= lhs->number_;
+    lhso = std::move(rhs);
+    return;
+  }
+  Ptr<UInt256Wrapper> n(
+      new UInt256Wrapper(vm_, fetch::vm::TypeIds::UInt256, lhs->number_ * rhs->number_));
+  lhso = std::move(n);
+}
+
+void UInt256Wrapper::InplaceMultiply(Ptr<Object> const &lhso, Ptr<Object> const &rhso)
+{
+  Ptr<UInt256Wrapper> lhs = lhso;
+  Ptr<UInt256Wrapper> rhs = rhso;
+  lhs->number_ *= rhs->number_;
+}
+
+void UInt256Wrapper::Divide(Ptr<Object> &lhso, Ptr<Object> &rhso)
+{
+  Ptr<UInt256Wrapper> lhs = lhso;
+  Ptr<UInt256Wrapper> rhs = rhso;
+  if (lhs->IsTemporary())
+  {
+    lhs->number_ /= rhs->number_;
```

Comment:
This should check for division by zero, and do a RuntimeError if so. I guess the other divide and inplace divisions need this check too.

Reply:
The zero division check is implemented inside UInt<> underlying container, and an exception is thrown in case of zero division there. 
ledger/libs/vectorise/include/vectorise/uint/uint.hpp:651
This class is basically just a wrapper above a concrete UInt<256> container. So I think an additional check here could be a redundant one.

Reply:
But RutimeError() has to be called here so that the VM can halt execution -- do you need to wrap the operation in try/catch perhaps?

Reply:
Ah I see, the vm_->RuntimeError("") must be called, not just an exception. Ok, I'll do the check here then of course.

Reply:
done